src/ReportExporter.php

```diff
@@ -97,18 +89,16 @@ public static function queue_report_export( $export_id, $report_type, $report_ar
 		$total_rows  = $exporter->get_total_rows();
 		$batch_size  = $exporter->get_limit();
 		$num_batches = (int) ceil( $total_rows / $batch_size );
-		$start_time  = time() + 5;
 
 		// Create batches, like initial import.
 		$report_batch_args = array( $export_id, $report_type, $report_args );
 
 		if ( 0 < $num_batches ) {
-			ReportsSync::queue_batches( 1, $num_batches, self::REPORT_EXPORT_ACTION, $report_batch_args );
+			self::queue_batches( 1, $num_batches, 'export_report', $report_batch_args );
 
 			if ( $send_email ) {
 				$email_action_args = array( get_current_user_id(), $export_id, $report_type );
-
-				ReportsSync::queue_dependent_action( self::REPORT_EXPORT_EMAIL_ACTION, $email_action_args, self::REPORT_EXPORT_ACTION );
+				self::queue_batches( 1, $num_batches, 'email_report_download_link', $email_action_args );
```

Comment:
This needs to get reverted to a single dependent job. The batches running `export_report` create the CSV, and then after we should send a single email.

As this currently sits, it creates several unnecessary jobs that all produce no emails (because of the unexpected batch arguments).

Reply:
You're right.  Updated this and the dependency (which was inverted) in https://github.com/woocommerce/woocommerce-admin/pull/3285/commits/45e2c8f08d817e9e74c75d69b39e2e5a617ed9aa